Assets/MRTK/Core/Definitions/SpatialAwareness/SpatialAwarenessPlanarObject.cs

```diff
@@ -12,32 +12,57 @@ public partial class SpatialAwarenessPlanarObject : BaseSpatialAwarenessObject
         /// </summary>
         public BoxCollider Collider { get; set; }
 
+        private SpatialAwarenessSurfaceTypes planeType = SpatialAwarenessSurfaceTypes.Unknown;
+
+        /// <summary>
+        /// The type of surface (ex: wall) represented by this object. 
+        /// </summary>
+        public SpatialAwarenessSurfaceTypes SurfaceType
+        {
+            get => planeType;
+            private set => planeType = value;
+        }
+
         /// <summary>
         /// Constructor.
         /// </summary>
-        public SpatialAwarenessPlanarObject() : base() { }
+        private SpatialAwarenessPlanarObject() : base() { }
 
         /// <summary>
         /// Creates a <see cref="SpatialAwarenessPlanarObject"/>.
         /// </summary>
         /// <returns>
         /// SpatialAwarenessPlanarObject containing the fields that describe the plane.
         /// </returns>
-        public static SpatialAwarenessPlanarObject CreateSpatialObject(Vector3 size, int layer, string name, int planeId)
+        public static SpatialAwarenessPlanarObject CreateSpatialObject(
+            Vector3 center,
+            Vector3 size,
+            Quaternion rotation,
+            int layer, 
+            string name, 
+            int planeId,
+            SpatialAwarenessSurfaceTypes surfaceType = SpatialAwarenessSurfaceTypes.Unknown)
         {
-            SpatialAwarenessPlanarObject newMesh = new SpatialAwarenessPlanarObject();
+            SpatialAwarenessPlanarObject newPlane = new SpatialAwarenessPlanarObject();
+
+            newPlane.Id = planeId;
+            newPlane.SurfaceType = surfaceType;
+
 
-            newMesh.Id = planeId;
-            newMesh.GameObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
-            newMesh.GameObject.name = name;
-            newMesh.GameObject.layer = layer;
-            newMesh.GameObject.transform.localScale = size;
+            GameObject planeObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            planeObject.transform.position = center;
+            planeObject.transform.rotation = rotation;
+            planeObject.transform.localScale = size;
+            planeObject.name = name;
+            planeObject.layer = layer;
 
-            newMesh.Filter = newMesh.GameObject.GetComponent<MeshFilter>();
-            newMesh.Renderer = newMesh.GameObject.GetComponent<MeshRenderer>();
-            newMesh.Collider = newMesh.GameObject.GetComponent<BoxCollider>();
+            newPlane.GameObject = planeObject;
+            newPlane.Filter = newPlane.GameObject.GetComponent<MeshFilter>();
+            newPlane.Renderer = newPlane.GameObject.GetComponent<MeshRenderer>();
+            newPlane.Renderer.shadowCastingMode = UnityEngine.Rendering.ShadowCastingMode.Off;
```

Comment:
This feels potentially implementation-specific

Reply:
by default, shadows off are a perf benefit.